typing/env.ml

```diff
@@ -2456,6 +2464,37 @@ and lookup_functor_components ~errors ~use ~loc lid env =
   | Error (No_components_alias p) ->
       may_lookup_error errors loc env (Cannot_scrape_alias(lid, p))
 
+and lookup_apply ~errors ~use ~loc lid0 env =
+  let lid_f0, args0 = decompose_apply lid0 in
+  let path_f0, comp_f0 =
+    lookup_module_components ~errors ~use ~loc lid_f0 env in
+  let check_one_apply ~errors ~use ~loc (lid,comp) lid_arg env =
```

Comment:
Why the tuple?

Reply:
To group together all the information regarding the functor to apply. In practice, that API is much less error prone than the previous one where it's a mixmatch of longidents and other information.